rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -122,9 +122,56 @@ local function sync_backends()
   end
 end
 
+local function route_to_alternative_balancer(balancer)
+  if not balancer.alternative_backends then
+    return false
+  end
+
+  -- TODO: support traffic shaping for n > 1 alternative backends
+  local alternative_balancer = balancers[balancer.alternative_backends[1]]
+
+  local clean_target_header = util.replace_special_char(alternative_balancer.traffic_shaping_policy.header, "-", "_")
+
+  local header = ngx.var["http_" .. clean_target_header]
+  if header then
+    if header == "always" then
+      return true
+    elseif header == "never" then
+      return false
+    end
+  end
+
+  local clean_target_cookie = util.replace_special_char(alternative_balancer.traffic_shaping_policy.cookie, "-", "_")
+
+  local cookie = ngx.var["cookie_" .. clean_target_cookie]
+  if cookie then
+    if cookie == "always" then
+      return true
+    elseif cookie == "never" then
+      return false
+    end
+  end
+
+  -- TODO: this function has a left-hand bias which produces
+  -- slightly smaller real weighting than ancitipated
+  if math.random(0,100) <= alternative_balancer.traffic_shaping_policy.weight then
```

Comment:
This seems incorrect to me - when I set weight to `0`, that should mean no traffic to alternative backend. However with this implementation there 1/101 chance that it will send traffic to the canary backend.

I think it should be `math.random(1, 100)`.

--

> this function has a left-hand bias which produces

Are you referring to `math.random`? Can you explain more where does that bias come from?